### PR TITLE
feat: return the sourceDocument when asking for documents info

### PR DESCRIPTION
**Pull request by romansharapov19**
**What type of PR is this?**
Feature: a new feature

**What this PR does / why we need it:**

- remove agents
- create a documentChain
- save the question and the answer when asking for document info
- return the document source when asking for document info

### CODE REVIEW

File: packages/api/src/ai/facades/ai.service.ts [hunk 1]
```diff
@@ -17,7 +16,7 @@ import { Document } from 'langchain/document';
 import { PromptTemplate } from 'langchain/prompts';
 import { BaseMessage, ChainValues } from 'langchain/schema';
 
-type AIExecutor = AgentExecutor | ConversationChain;
+type AIExecutor = DocumentConversationChainService | ConversationChain;
```
Comment (comoser) [inline packages/api/src/ai/facades/ai.service.ts:19]:
```suggestion
type AIExecutor = DocumentConversationChain | ConversationChain;
```

File: packages/api/src/ai/services/document-conversation-chain.service.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    private readonly memoryService: MemoryService,
+    private readonly vectorDbService: VectorDbService,
+    private readonly simpleConversationChainService: SimpleConversationChainService,
+    private readonly meta: DocumentConversationChainServiceProps
+  ) {
+    this.defaultChatPrompt = `Use the following pieces of context to answer the question at the end.
+                              {context}
+                              Question: {question}
+                              If you don't know the answer return only: Answer not found.
+                              But if you have an answer provide the most detailed response you can.`;
```
Comment (comoser) [inline packages/api/src/ai/services/document-conversation-chain.service.ts:30]:
```suggestion
    this.defaultChatPrompt = `Use the following context to answer the question at the end.
                              {context}
                              Question: {question}
                              If you don't know the answer return only: Answer not found.
                              But if you have an answer, provide the most detailed response you can.`;
```

File: packages/api/src/ai/services/memory.service.ts [hunk 3]
```diff
@@ -70,4 +70,26 @@ export class MemoryService {
       })
     );
   }
+
+  async createMemoryWithDocumentInput(
+    roomId: string,
+    input: string,
+    response: string
+  ) {
+    const redisDocumentResponse = new RedisChatMessageHistory({
```
Comment (comoser) [inline packages/api/src/ai/services/memory.service.ts:79]:
```suggestion
    const redisChatHistory = new RedisChatMessageHistory({
```

Comment (comoser) [reply packages/api/src/ai/services/memory.service.ts:79]: also, wouldn't it be better to fetch the memory in cache before creating a new one? just leverage the existing methods in memory service

File: packages/api/src/chats/usecases/upload-documents-to-chat.usecase.ts [hunk 4]
```diff
@@ -59,8 +59,6 @@ export class UploadDocumentsToChatUsecase implements Usecase {
         )
       );
     }
-
-    this.aiService.invalidateAgentCache(roomId);
```
Comment (comoser) [inline packages/api/src/chats/usecases/upload-documents-to-chat.usecase.ts:63]: This needs to be tested out with the following test:
- create room A
- upload doc A
- ask question about doc A
- upload doc B
- ask question about doc B

If everything is correct, then all is good.

Comment (romansharapov19) [reply packages/api/src/chats/usecases/upload-documents-to-chat.usecase.ts:63]: Already tested it, seems to work as expected.

**Review by comoser — APPROVED** [review]
amazing 👍 🚀

**Pull request merged**